### PR TITLE
JAV-469 Edge lb new instance cache manager

### PR DESCRIPTION
**Pull request by wujimin**
edge use new InstanceCacheManager, normal consumer use old InstanceCacheManager
if there is no problem, then old InstanceCacheManager will be deleted.

### CODE REVIEW

File: core/src/main/java/io/servicecomb/core/definition/classloader/DefaultMicroserviceClassLoaderFactory.java [hunk 1]
```diff
@@ -20,7 +20,7 @@
   public static final MicroserviceClassLoaderFactory INSTANCE = new DefaultMicroserviceClassLoaderFactory();
 
   @Override
-  public ClassLoader create(String microserviceName, String version) {
+  public ClassLoader create(String appId, String microserviceName, String version) {
     return Thread.currentThread().getContextClassLoader();
```
Comment (WillemJiang) [inline core/src/main/java/io/servicecomb/core/definition/classloader/DefaultMicroserviceClassLoaderFactory.java]: If we just return the TCC, maybe we need to update the method name for it.

File: edge/edge-core/src/main/resources/microservice.yaml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+cse-config-order: -500
```
Comment (WillemJiang) [inline edge/edge-core/src/main/resources/microservice.yaml]: Please add the license header here.

File: service-registry/src/main/java/io/servicecomb/serviceregistry/registry/AbstractServiceRegistry.java [hunk 3]
```diff
@@ -95,6 +98,36 @@ public void init() {
     eventBus.register(this);
   }
 
+  protected void initAppManager() throws InstantiationException, IllegalAccessException, ClassNotFoundException {
+    appManager = new AppManager(eventBus);
+
+    String microserviceVersionFactoryClass = serviceRegistryConfig.getMicroserviceVersionFactory();
+    if (microserviceVersionFactoryClass == null) {
+      return;
```
Comment (WillemJiang) [inline service-registry/src/main/java/io/servicecomb/serviceregistry/registry/AbstractServiceRegistry.java]: It's better to throw exception here.

File: core/src/main/java/io/servicecomb/core/definition/classloader/DefaultMicroserviceClassLoaderFactory.java (hunk 1 above)
Comment (wujimin) [reply core/src/main/java/io/servicecomb/core/definition/classloader/DefaultMicroserviceClassLoaderFactory.java]: this just a Default factory, another PrivateXXXFactory will do a real create

File: service-registry/src/main/java/io/servicecomb/serviceregistry/registry/AbstractServiceRegistry.java (hunk 3 above)
Comment (wujimin) [reply service-registry/src/main/java/io/servicecomb/serviceregistry/registry/AbstractServiceRegistry.java]: in this version, we did not remove old cacheManager, and new cacheManager only used by edge.
so microserviceVersionFactoryClass is null, means not work with edge, will use old cacheManager, can not throw exception

Comment (WillemJiang) [reply service-registry/src/main/java/io/servicecomb/serviceregistry/registry/AbstractServiceRegistry.java]: Can we add a comment to the code for it?

Comment (wujimin) [reply service-registry/src/main/java/io/servicecomb/serviceregistry/registry/AbstractServiceRegistry.java]: done.

File: edge/edge-core/src/main/resources/microservice.yaml (hunk 2 above)
Comment (wujimin) [reply edge/edge-core/src/main/resources/microservice.yaml]: discussed, no need to add license here.

**Pull request merged**